src/other/codemirror/utils/exaxyz_linter.ts

```diff
@@ -0,0 +1,50 @@
+import { Diagnostic } from "@codemirror/lint";
+import { Annotation, StateField, Transaction } from "@codemirror/state";
+import { EditorView } from "@codemirror/view";
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { ConsistencyChecks } from "@exabyte-io/code.js/src/types";
+import _ from "underscore";
+
+export const ChecksAnnotation = Annotation.define<{ checks: ConsistencyChecks }>();
+
+export const checksStateField = StateField.define({
+    create(): ConsistencyChecks {
+        return { keys: [], messages: [] };
+    },
+    update: (value: ConsistencyChecks, tr: Transaction) => {
```

Comment:
should be returned value type defined there?

Reply:
Also I'd suggest to have full names for params (no matter we use TS or not) better for readability to have `transaction` not `tr`